pandas/_testing/_warnings.py

```diff
@@ -99,13 +107,32 @@ class for all warnings. To raise multiple types of exceptions,
             yield w
         finally:
             if expected_warning:
-                expected_warning = cast(type[Warning], expected_warning)
-                _assert_caught_expected_warning(
-                    caught_warnings=w,
-                    expected_warning=expected_warning,
-                    match=match,
-                    check_stacklevel=check_stacklevel,
-                )
+                if isinstance(expected_warning, tuple) and must_find_all_warnings:
+                    match = (
+                        match
+                        if isinstance(match, tuple)
+                        else tuple(match for i in range(len(expected_warning)))
+                    )
+                    for warning_type, warning_match in zip(expected_warning, match):
+                        _assert_caught_expected_warning(
+                            caught_warnings=w,
+                            expected_warning=warning_type,
+                            match=warning_match,
+                            check_stacklevel=check_stacklevel,
+                        )
+                else:
+                    expected_warning = cast(type[Warning], expected_warning)
```

Comment:
Yes, you're right. I believe this was originally done, to suppress `_assert_caught_expected_warning` not accepting tuples. I changed that and fixed some issues that arose because of that now.